From 619da21b72a9d97f52128546889ba4c9beb27305 Mon Sep 17 00:00:00 2001
From: Isaac Davis <isaac.davis@joyent.com>
Date: Mon, 26 Jun 2017 20:55:28 +0000
Subject: [PATCH] OS-6203 VM.lookup breaks if "options" argument is empty
 Reviewed by: Dave Eddy <dave.eddy@joyent.com> Reviewed by: Josh Wilsdon
 <josh@wilsdon.ca> Approved by: Dave Eddy <dave.eddy@joyent.com>

---
 src/vm/node_modules/VM.js | 12 +++++-------
 1 file changed, 5 insertions(+), 7 deletions(-)

diff --git a/src/vm/node_modules/VM.js b/src/vm/node_modules/VM.js
index d4c023f4..2f1c8009 100644
--- a/src/vm/node_modules/VM.js
+++ b/src/vm/node_modules/VM.js
@@ -1476,7 +1476,6 @@ exports.lookup = function (search, options, callback)
     var key;
     var matches;
     var need_fields = [];
-    var results = [];
     var tracers_obj;
     var transform;
 
@@ -1559,8 +1558,7 @@ exports.lookup = function (search, options, callback)
 
     lookup_opts = {log: log, fields: need_fields};
     vmload.getVmobjs(lookupFilter, lookup_opts, function gotVMs(err, vmobjs) {
-        var r;
-        var short_results = [];
+        var results;
 
         if (err) {
             callback(err);
@@ -1589,10 +1587,10 @@ exports.lookup = function (search, options, callback)
                 }
             }));
         } else {
-            for (r in vmobjs) {
-                short_results.push(results[r].uuid);
-            }
-            callback(null, short_results);
+            results = vmobjs.map(function getUUID(vmobj) {
+                return vmobj.uuid;
+            });
+            callback(null, results);
         }
     });
 };
-- 
2.21.0

